Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -324,6 +329,8 @@ extension BrowserViewController: WKNavigationDelegate {
         return (.cancel, preferences)
       }
 
+      tab?.braveSearchResultAdManager = BraveSearchResultAdManager(url: requestURL, rewards: rewards, isPrivateBrowsing: isPrivateBrowsing)
```

Comment:
What if `tab?.loadRequest(modifiedRequest)` is called on line 328. I know that `BraveSearchResultAdManager` returns `nil` for Brave Rewards users, however it may not be clear to others in the future if/when these events are sent when ads are enabled.

Reply:
if `tab?.loadRequest(modifiedRequest)` is called, then we will run this code again but with `X-Brave-Ads-Enabled` header. `tab?.braveSearchResultAdManager` will be nil because rewards are enabled. Seems like this works as expected.